### ISSUE TITLE
[Question] Missing tests for the application concerns

### ISSUE THREAD

**Issue by minhnimble** [M-I-N/NimbleSurvey#11]
## Issue

Regarding the #8 issue, thank you for letting me know in advance of your current knowledge about unit testing. Unfortunately, writing unit tests is a crucial part of the evaluation and at this point, it is clear to see that there is no unit test or UI test found for your application. Hence, if you are requested to add some Unit/UI tests to help improve the code coverage for this project (above 50%) for all the logic code used in the application, what is your plan to achieve that requirement? 

Apart from that, I would like to know deeper on what is an ideal mentorship or environment to you for learning about writing unit tests?

**Comment by M-I-N** [comment]
I agree to the fact that testing is a crucial part of any software. And to be honest, I was in hesitation whether I should start the development of the assignment or not. Because it was clearly stated in the assignment that the unit tests should be covered. Also I communicated about not having my experience in unit testing in the very first interview. But still I chose to start working in this assignment because I think no effort is wasted effort. Doesn't matter if I'm considered a fit or not for this position. In the end I will gain some knowledge if not a lot. 

---

Speaking about my plan to achieve the code coverage with Unit/UI tests, I think I'll need help from an experienced person may they be from the team I'll be working with or any mentorship program available in the industry. Besides, I have had a long desire to learn TDD though it’s long overdue. 

> Unit/UI testing is not very common in the culture of software development in my country. Most employers don't welcome these practices in general because they think these are wastage of time, effort and money. 

As a result I fall short in learning this skill.
